core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/dr/BackupRestoreSaltConfigGenerator.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.cloudbreak.reactor.handler.cluster.dr;
+
+import com.sequenceiq.cloudbreak.orchestrator.model.SaltConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.SaltPillarProperties;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.stereotype.Component;
+
+import static java.util.Collections.singletonMap;
+
+@Component
+public class BackupRestoreSaltConfigGenerator {
+
+    public SaltConfig createSaltConfig(String location, String backupId, String cloudPlatform) throws URISyntaxException {
+        String fullLocation = buildFullLocation(location, backupId, cloudPlatform);
+
+        Map<String, SaltPillarProperties> servicePillar = new HashMap<>();
+        servicePillar.put("disaster-recovery", new SaltPillarProperties("/postgresql/disaster_recovery.sls",
+            singletonMap("disaster_recovery", singletonMap("object_storage_url", fullLocation))));
+        return new SaltConfig(servicePillar);
+    }
+
+    public String buildFullLocation(String location, String backupId, String cloudPlatform) throws URISyntaxException {
+        URI uri = new URI(location);
+        String fullLocation;
+        if ("aws".equalsIgnoreCase(cloudPlatform)) {
+            if (uri.getScheme() != null && !uri.getScheme().startsWith("s3")) {
+                throw new UnsupportedOperationException("Incorrect URL scheme for AWS cloud platform: " + location);
```

Comment:
I think validating this here is a bit late. This should be caught when processing the request. Just check how this works for logging/telemetry

Reply:
Where is the logging/telemetry logic? This is my first experience in this codebase so I'm not sure exactly what you're referencing.

Reply:
so the validation should be done around `DatabaseBackupRestoreService#backupDatabase/restoreDatabase` and the flow shouldn't be started with invalid input
if you check `LoggingBase` it has separate input for AWS/Azure and those have validators, so it's checked on request time

Reply:
I've added a `BackupRestoreV4RequestValidator` class and unit tests for it. It checks the location scheme, and also makes sure that required parameters are set.